src/Tribe/Views/V2/Views/Traits/HTML_Cache.php

```diff
@@ -196,4 +200,152 @@ protected function should_enable_html_cache( $context ) {
+		 *
+		 * @param array   $nonces   List of action and field name where the nonce is stored.
+		 * @param Context $context  Context from the current view.
+		 * @param View    $view     Which view instance we are currently rendering.
+		 */
+		return apply_filters( 'tribe_events_views_v2_get_view_nonce_fields', $nonces, $this->get_context(), $this );
+	}
+
+	/**
+	 * Get the list of attributes we will do replacement for HTML Cache.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return array   List of attributes that we are going to replace.
+	 */
+	protected function get_view_nonce_attributes() {
+		$nonces = [
+			'wp_rest' => 'data-view-rest-nonce',
+		];
+
+		/**
+		 * Filter to control nonce attributes replacement for HTML Cache.
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param array   $nonces   List of action and field name where the nonce is stored.
+		 * @param Context $context  Context from the current view.
+		 * @param View    $view     Which view instance we are currently rendering.
+		 */
+		return apply_filters( 'tribe_events_views_v2_get_view_nonce_attributes', $nonces, $this->get_context(), $this );
+	}
+
+	/**
+	 * Get the list of JSON properties we will do replacement for HTML Cache.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return array   List of json properties that we are going to replace.
+	 */
+	protected function get_view_nonce_json_properties() {
+		$nonces = [
+			'wp_rest' => 'rest_nonce',
+		];
+
+		/**
+		 * Filter to control nonce json properties replacement for HTML Cache.
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param array   $nonces   List of action and field name where the nonce is stored.
+		 * @param Context $context  Context from the current view.
+		 * @param View    $view     Which view instance we are currently rendering.
+		 */
+		return apply_filters( 'tribe_events_views_v2_get_view_nonce_json_properties', $nonces, $this->get_context(), $this );
+	}
+
+	/**
+	 * Does string replacement on the HTML cached based on the possible places we look for cached nonce values to inject
+	 * the correct string placeholder so we can remove it later.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param string   $html  HTML with the nonces to be replaced.
+	 *
+	 * @return string  HTML after replacement is complete.
+	 */
+	protected function extract_nonces_before_cache( $html ) {
+		$nonce_fields = $this->get_view_nonce_fields();
+		$nonce_attrs  = $this->get_view_nonce_attributes();
+		$nonce_props  = $this->get_view_nonce_json_properties();
+
+		foreach ( $nonce_fields as $action => $field ) {
+			$html = preg_replace( '!(<input[^>]+name="' . preg_quote( $field ) . '"[^>]+value=")[^"]*("[^>]*>)!', '\1%%NONCE:' . $action . '%%\2', $html );
+			$html = preg_replace( '!(<input[^>]+value=")[^"]*("[^>]+name="' . preg_quote( $field ) . '"[^>]*>)!', '\1%%NONCE:' . $action . '%%\2', $html );
+		}
+
+		foreach ( $nonce_attrs as $action => $attr ) {
+			$html = preg_replace( '!(' . preg_quote( $attr ) . '=")[^"]*(")!', '\1%%NONCE:' . $action . '%%\2', $html );
```

Comment:
:warning: Passing the $delimiter as the second parameter to preg_quote() is strongly recommended.

<sub>hash: e398915e5b83b739b41fe00fa484419c</sub>